test/python/http_recorder.py

```diff
@@ -0,0 +1,285 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Utilities (based on VCRpy) to record remote requets and allow testing offline/cached."""
+
+import json
+from vcr.persisters.filesystem import FilesystemPersister
+from vcr import VCR
+
+
+class IdRemoverPersister(FilesystemPersister):
+    """
+    IdRemoverPersister is a VCR persister. This is, it implements a way to save and load cassettes.
+    This persister in particular inherits load_cassette from FilesystemPersister (basically, it
+    loads a standard cassette in the standard way from the FS). On the saving side, it removes some
+    fields in the JSON content of the responses for dummy values.
+    """
+
+    @staticmethod
+    def get_responses_with(string2find, cassette_dict):
+        """
+        Filters the requests from cassette_dict
+
+        Args:
+            string2find (str): request path
+            cassette_dict (dict): a VCR cassette dictionary
+
+        Returns:
+            Request: VCR's representation of a request.
+        """
+        requests_indeces = [i for i, x in enumerate(cassette_dict['requests']) if
+                            string2find in x.path]
+        return [cassette_dict['responses'][i] for i in requests_indeces]
+
+    @staticmethod
+    def get_new_id(field, path, id_tracker, _type=str):
+        """
+        Creates a new dummy id (or value) for replacing an existing id (or value).
+
+        Args:
+            field (str): field name is used, in same cases, to create a dummy value.
+            path (str): path of the request is used, in same cases, to create a dummy value.
+            id_tracker (dict): a map of already assigned ids and generated ids.
+            _type (type): type of the value.
+
+        Returns:
+            str: that is used to replace a value.
+        """
+
+        if _type == float:
+            return 0.42
+        if _type == int:
+            return 42
+        dummy_name = 'dummy%s%s' % (path.replace('/', ''), field)
+        count = len(list(filter(lambda x: str(x).startswith(dummy_name), id_tracker.values())))
+        return "%s%02d" % (dummy_name, count + 1)
+
+    @staticmethod
+    def get_maching_dicts(data_dict, map_list):
+        """
+        Find subdicts that are described in map_list.
+        Args:
+            data_dict (dict): in which the map_list is going to be searched.
+            map_list (list): the list of nested keys to find in the data_dict
+
+        Returns:
+            dict: the dictionary in which matches map_list.
```

Comment:
According to the code, this is returning a list. Can you update the docstring accordingly?

Reply:
5e7bf25